### PR TITLE
Console host fixes

### PR DESCRIPTION
**Pull request by mathewc**
Recent changes broke the self host console scenario

### CODE REVIEW

File: src/WebJobs.Script/Host/ScriptHost.cs [hunk 1]
```diff
@@ -558,7 +558,7 @@ private void PreInitialize()
             // taking this snapshot once and reusing at various points during initialization allows us to
             // minimize disk operations
             _directorySnapshot = Directory.EnumerateDirectories(ScriptConfig.RootScriptPath).ToImmutableArray();
-            TraceWriter.Verbose("Created directory snapshot.");
+            TraceWriter?.Verbose("Created directory snapshot.");
```
Comment (mathewc) [inline src/WebJobs.Script/Host/ScriptHost.cs]: In normal Azure Functions scenarios, the TraceWriter will be defaulted before the host is created, so will be available early in the pipeline. In SelfHost, that is not the case.

Comment (pragnagopa) [reply src/WebJobs.Script/Host/ScriptHost.cs]: can you add a test to verify scripthost initialization works when self hosted?

File: test/WebJobs.Script.Tests.Integration/Host/ScriptHostManagerTests.cs [hunk 2]
```diff
@@ -577,15 +605,11 @@ public void IsHostHealthy_ReturnsExpectedResult()
             hostManager.Stop();
             Assert.Equal(ScriptHostState.Default, hostManager.State);
 
-            await Task.Delay(FileTraceWriter.LogFlushIntervalMs);
-
-            string hostLogFilePath = Directory.EnumerateFiles(Path.Combine(logDir, "Host")).Single();
-            string hostLogs = File.ReadAllText(hostLogFilePath);
```
Comment (mathewc) [inline test/WebJobs.Script.Tests.Integration/Host/ScriptHostManagerTests.cs]: Fixing flakiness I was seeing in file access. Simplifying test to get logs from the TestTraceWriter instead

File: src/WebJobs.Script/Host/ScriptHost.cs [hunk 3]
```diff
@@ -800,6 +800,11 @@ private void ConfigureLogging()
                 .Filter(p => { return true; })
                 .Subscribe(HandleHostError);
 
+            if (TraceWriter != null)
+            {
+                TraceWriter = TraceWriter.WithDefaults(ScriptConstants.TraceSourceScriptHost, _instanceId);
```
Comment (pragnagopa) [inline src/WebJobs.Script/Host/ScriptHost.cs]: Nit: change to TraceWriter?.WithDefaults

**Pull request merged**